src/node_persistent.h

```diff
@@ -0,0 +1,27 @@
+#ifndef SRC_NODE_PERSISTENT_H_
+#define SRC_NODE_PERSISTENT_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "v8.h"
+
+namespace node {
+
+template <typename T>
+struct ResetInDestructorPersistentTraits {
```

Comment:
A key reason leaks end up happening around all this is the fact that much of this is horribly under-documented.  Mind adding some code comments in here that explains how/why this is here?

Reply:
Good idea, I'll do that.